packages/devtools_app/lib/src/screens/memory/panes/diff/controller/diff_pane_controller.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../../../primitives/utils.dart';
+import '../../../../../shared/common_widgets.dart';
+import '../../../primitives/memory_utils.dart';
+import '../../../shared/heap/model.dart';
+import 'model.dart';
+
+class DiffPaneController {
+  DiffPaneController(this.snapshotTaker);
+
+  final listScrollController = ScrollController();
```

Comment:
scroll controllers should be created in widget lifecycle methods, preferably initState

Reply:
The button that invokes the method to add snapshot lives in _ListControlPane, while the list that should be scrolled lives in _SnapshotListItems. Both are stateless. 
Should I convert list widget to stateful and add notifier to the controller, to listen for signal to scroll to bottom?

Sounds as unnecessary complexity.